distributed/client.py

```diff
@@ -648,15 +648,24 @@ def __init__(
                 "Unexpected keyword arguments: {}".format(str(sorted(kwargs)))
             )
 
+        from .deploy import Cluster
+
         if isinstance(address, (rpc, PooledRPCCall)):
             self.scheduler = address
-        elif hasattr(address, "scheduler_address"):
+        elif isinstance(address, Cluster):
             # It's a LocalCluster or LocalCluster-compatible object
             self.cluster = address
             with suppress(AttributeError):
                 loop = address.loop
             if security is None:
                 security = getattr(self.cluster, "security", None)
+        elif address is not None and not isinstance(address, str):
+            if issubclass(address, Cluster):
```

Comment:
The second condition seems to imply the first.  Also, the second instance seems very specific.  I wonder if maybe this could be more broad like the following:

```python
elif not isinstance(address, str):
    raise ValueError("Scheduler address must be a string or a Cluster instance, got {}".format(address))
```

Reply:
thank you for the feedback. I made the error message more broad, as suggested, but had to also check that `address is not None` otherwise this would raise an error when address was None which caused some tests to fail. I'm not sure if it still makes sense for `address` to be None at this point in the setup, but I just preserved the behavior that was expected by the failing tests.